packages/manager/apps/telecom/src/app/telecom/pack/migration/meeting/pack-migration-meeting.controller.js

```diff
@@ -0,0 +1,135 @@
+import get from 'lodash/get';
+
+export default class TelecomPackMigrationMeetingCtrl {
+  /* @ngInject */
+  constructor(
+    $scope,
+    $translate,
+    OvhApiConnectivityEligibility,
+    OvhApiConnectivityEligibilitySearch,
+    TucPackMigrationProcess,
+    TucToast,
+  ) {
+    this.$scope = $scope;
+    this.$translate = $translate;
+    this.OvhApiConnectivityEligibility = OvhApiConnectivityEligibility;
+    this.OvhApiConnectivityEligibilitySearch = OvhApiConnectivityEligibilitySearch;
+    this.TucPackMigrationProcess = TucPackMigrationProcess;
+    this.TucToast = TucToast;
+  }
+
+  $onInit() {
+    this.selectedMeeting = {
+      meetingSlots: {
+        fakeMeeting: false,
+        slot: null,
+      },
+    };
+    this.contactName = null;
+    this.showMeetingSlots = false;
+    this.meetingSlots = {};
+
+    this.process = this.TucPackMigrationProcess.getMigrationProcess();
+    this.buildingReference = this.process.selectedOffer.buildingReference;
+
+    this.loading = true;
+    this.meetings = [];
+    return this.OvhApiConnectivityEligibility.v6()
+      .testBuilding(this.$scope, { building: this.buildingReference })
+      .then((eligibility) => {
+        if (eligibility.result) {
+          const { eligibilityReference } = eligibility.result;
+          const productCode = eligibility.result.offers[0].product.code;
+          this.process.selectedOffer.address =
+            eligibility.result.endpoint.address;
+          return this.OvhApiConnectivityEligibilitySearch.v6()
+            .searchMeetings(this.$scope, {
+              eligibilityReference,
+              productCode,
+            })
+            .then((data) => {
+              if (data.result) {
+                this.meetingSlots.canBookFakeMeeting =
+                  data.result.canBookFakeMeeting;
+                this.meetingSlots.slots = data.result.meetingSlots;
+
+                let slots = [];
+                let prevTitle;
+                data.result.meetingSlots.forEach((slot, index) => {
+                  const title = moment(slot.startDate).format(
+                    'ddd DD MMM YYYY',
+                  );
+                  if (!prevTitle) {
+                    prevTitle = title;
+                  } else if (prevTitle !== title) {
+                    this.meetings.push({
+                      title: prevTitle,
+                      slots,
+                    });
+                    slots = [];
+                    prevTitle = title;
+                  }
```

Comment:
**question:** Can you please explain the need for this piece of code. If I understand well, you're trying to add meetingSlots (ignoring the first option) but with title set as previous slot's formatted startDate.

Is my understanding correct? What is the requirement?

Reply:
From the API call, we receive a block of available meeting dates. This piece of code helps us to manage meeting date like for example :
- receive from the API : [{startDate: '2023-03-15 08:00:00', endDate: '2023-03-15 12:00:00'},{startDate: '2023-03-16 08:00:00', endDate: '2023-03-16 12:00:00'},{startDate: '2023-03-17 08:00:00', endDate: '2023-03-17 12:00:00'},{startDate: '2023-03-17 14:00:00', endDate: '2023-03-17 18:00:00'}....]
- to display : [[{startDate: '2023-03-15 08:00:00', endDate: '2023-03-15 12:00:00'}], [{startDate: '2023-03-16 08:00:00', endDate: '2023-03-16 12:00:00'}], [{startDate: '2023-03-17 08:00:00', endDate: '2023-03-17 12:00:00'},{startDate: '2023-03-17 14:00:00', endDate: '2023-03-17 18:00:00'}]....]

And the title corresponding to the date to display